h2o-core/src/main/java/hex/Model.java

```diff
@@ -1251,13 +1280,7 @@ else if (isWeights && computeMetrics) {
             vec = evec;
           }
         } else if(vec.isCategorical()) {
-          if (parms._categorical_encoding == Parameters.CategoricalEncodingScheme.LabelEncoder) {
-            Vec evec = vec.toNumericVec();
-            toDelete.put(evec._key, "label encoded vec");
-            vec = evec;
-          } else {
-            throw new IllegalArgumentException("Test/Validation dataset has categorical column '" + names[i] + "' which is real-valued in the training data");
-          }
+          throw new IllegalArgumentException("Test/Validation dataset has categorical column '" + names[i] + "' which is real-valued in the training data");
```

Comment:
@michalk: decided to completely remove this fallback as it should not be necessary anymore, all conversion should be handled explicitly with logic above.